clients/src/test/java/org/apache/kafka/common/security/authenticator/SaslAuthenticatorTest.java

```diff
@@ -1550,13 +1549,8 @@ public void testCannotReauthenticateWithDifferentPrincipal() throws Exception {
          * original token with a new one.
          */
         delay(1000L);
-        try {
-            checkClientConnection(node);
-            fail("Re-authentication with a different principal should have failed but did not");
-        } catch (AssertionError e) {
-            // ignore, expected
-            server.verifyReauthenticationMetrics(0, 1);
-        }
+        assertThrows(AssertionFailedError.class, () -> checkClientConnection(node));
```

Comment:
It would be better that we can make accurate assert for ```checkClientConnection```.

Reply:
We can file a JIRA if you think this can be improved. We didn't change that here, right?

Reply:
trace it by https://issues.apache.org/jira/browse/KAFKA-12188